src/xo-app/jobs/scheduling/index.js

```diff
@@ -0,0 +1,193 @@
+import ActionButton from 'action-button'
+import find from 'lodash/find'
+import isEmpty from 'lodash/isEmpty'
+import map from 'lodash/map'
+import React, { Component } from 'react'
+import { error } from 'notification'
+import Icon from 'icon'
+import Scheduler, { SchedulePreview } from 'scheduling'
+import { Toggle } from 'form'
+import { SelectPlainObject } from '../helpers'
+
+import {
+  createSchedule,
+  deleteSchedule,
+  subscribeJobs,
+  subscribeSchedules,
+  updateSchedule
+} from 'xo'
+
+const JOB_KEY = 'genericTask'
+
+export default class Schedules extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      action: undefined,
+      actions: undefined,
+      job: undefined,
+      jobs: undefined,
+      cron: '* * * * *'
+    }
+    this.loaded = new Promise((resolve, reject) => {
+      this._resolveLoaded = resolve
+    })
+      .then(() => {
+        const { id } = this.props
+        if (id) {
+          this._edit(id)
+        }
+      })
+  }
+
+  componentWillMount () {
+    const unsubscribeJobs = subscribeJobs(jobs => {
+      const j = {}
+      for (const id in jobs) {
+        const job = jobs[id]
+        if (job && (job.key === JOB_KEY)) {
+          const _job = {...job}
+          _job.label = `${_job.name} - ${_job.method} (${_job.id})`
+          j[job.id] = _job
+        }
+      }
+      this.setState({jobs: j})
+    })
+
+    const unsubscribeSchedules = subscribeSchedules(schedules => {
+      const s = {}
+      const { jobs } = this.state
+      if (isEmpty(jobs)) {
+        return
+      }
+      for (const id in schedules) {
+        const schedule = schedules[id]
+        const scheduleJob = find(jobs, job => job.id === schedule.job)
+        if (scheduleJob && scheduleJob.key === JOB_KEY) {
+          s[id] = schedule
+        }
+        this.setState({schedules: s}, this._resolveLoaded)
+      }
+    })
+
+    this.componentWillUnmount = () => {
+      unsubscribeJobs()
+      unsubscribeSchedules()
+    }
+  }
```

Comment:
At first sight, I lose the opportunity to pass a cb as a 2nd parameter to setState.
Simple to add this to addSubscripiton, but in fact the callback I wished to pass is created in the Component's constructor, so it becomes tricky to use it in the prop map passed to @addSubscription.

I feel I won't use it in this case, as it will not really simplify my code...unless you see this kind of good ideas I usually don't see at first :)